frontend/components/filter/SortBy.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import { View, Text } from "react-native";
+import { StyleSheet } from "react-native";
+import { Button } from "@/components/Button";
+import { ArrowUpward, ArrowDownward } from "@/components/icons/Icons";
+import { TSortOption } from "@/types/filter";
+
+interface SortRowProps {
+    id: string;
+    selectedSort: TSortOption; // the title of the current row selected, if any and the direction
+    // Function to update the sort selection in the parent
+    onPress: () => void;
+}
+
+export function SortRow({ id: id, selectedSort, onPress }: SortRowProps) {
+    const getSubtitle = () => {
+        switch (selectedSort.direction) {
+            case "up":
+                return "Highest to lowest";
+            case "down":
+                return "Lowest to highest";
+            case "none":
+                return "-";
+        }
+    };
+
+    return (
+        <View style={styles.rowContainer}>
+            <View style={styles.textContainer}>
+                <Text style={styles.sortFieldTitle}>{id}</Text>
+                <Text style={styles.subTitle}>{getSubtitle()}</Text>
```

Comment:
Calling functions within render methods of react components is generally considered bad practice for performance reasons, the function ends up getting called every single re-render. I think we can use `useEffect` and `useState` instead to achieve the same effect -> Have the actual text be a piece of state and you can move the switch inside a `useEffect`